### PR TITLE
Fixed unsanctioned yaw unkill problem with uninitialised joysticks

### PR DESCRIPTION
**Pull request by kevindehecker**
Atleast the HobbyKing RC initializes often in such a way that all axis are maxed out. This causes the drone to repeatedly unkill (because of full yaw), and to dance around violently (because full pitch and roll). 

Today I got bitten twice by this problem. Luckily it cost me only 1 finger. This pull request at least prevents the unkill part. (which happens very unexpectedly, just after uploading from center).

### CODE REVIEW

**Comment by podhrmic** [comment]
That sounds pretty serious. Maybe we should add some warning against using HobbyKing RCs and recommend reliable ones such as Taranis or Turnigy? Do you know which model it was?

**Comment by kevindehecker** [comment]
It just says Hobbyking and a bunch of features (6 channels, USB, etc), no number. We at TUDelft have a bunch of those (I guess they are cheap).

**Comment by podhrmic** [comment]
can you post the link to the product so we can add a proper warning?

**Comment by kevindehecker** [comment]
http://www.hobbyking.com/hobbyking/store/__20951__Hobbyking_6CH_RC_Flight_Simulator_System_Mode_2_.html

**Comment by podhrmic** [comment]
wow, that is an incredible price...!

**Comment by kevindehecker** [comment]
Haha, and there I was mistaking your interest for concern :P

**Comment by podhrmic** [comment]
well the question is how much do you value your fingers...:-P

File: sw/airborne/firmwares/rotorcraft/autopilot_arming_yaw.h [hunk 1]
```diff
@@ -30,6 +30,7 @@
 #define AUTOPILOT_ARMING_YAW_H
 
 #include "autopilot_rc_helpers.h"
+#include "subsystems/radio_control/rc_datalink.h"
```
Comment (flixr) [inline sw/airborne/firmwares/rotorcraft/autopilot_arming_yaw.h]: We can't have any implementation specific radio control stuff in here...

**Comment by kevindehecker** [comment]
@flixr how about this then? (Tomorrow I will test on drone)

**Comment by flixr** [comment]
I think it would be better to add another state that is entered on startup and only allow transition to the `MOTORS_OFF` state if throttle is low and the yaw is centered...
In the end this really needs to be handled in input2ivy...

**Comment by kevindehecker** [comment]
OK good idea, in the state machine it is. While I was at it, also added a fix for #1010

(Untested, will do that hopefully tomorrow)

**Comment by flixr** [comment]
You at least need to set the inital state to STATUS_INITIALISE_RC as well....

**Comment by kevindehecker** [comment]
Tnx :)

**Comment by kevindehecker** [comment]
OK, it works on a drone.

One remark: if the sticks are not properly centered (e.g. because the joystick needs to be initialized, or calibrated), the motors won't start. Which I think is a good thing. But no feedback to the user is provided that the joystick is the culprit. (#1011)

**Comment by flixr** [comment]
I'm also not sure if we should require the roll and pitch sticks to be centred....

**Comment by kevindehecker** [comment]
I am strong proponent of that, simply because currently <i>all</i> the sticks need to be touched before they are initialized. Separately.

If you unkill with just an initialized yaw in ATT mode, but non-initialized pitch/roll, the drone will flip instantly.

**Comment by flixr** [comment]
Well, you are using a joystick, most people have a proper RC.
And then you seem to be the first one to actually use such a crappy joystick, I never had any problems with the 3 different ones I have ;-)

I just remember that we had this discussion quite some time ago and some people explicitly didn't want the requirement of having all sticks centred...
Only want to prevent people from wondering why the suddenly can't arm anymore... but maybe it is better anyway, because it forces people to at least roughly calibrate their input devices....

@gautierhattenberger do you remember that discussion? what do you think?

**Comment by kevindehecker** [comment]
I think  this is the kind of issue (chance of actual accidents) in which "yes the problem exist, but not always" is not a reason to ignore it.

(I do agree about the wondering part, but that is something that can should fixed in addition.)

**Comment by gautierhattenberger** [comment]
What I remember is that sometimes, the initial values returned by sdl are stupid until you have moved your joystick to its full positions. Which should be done on the ground before powering the plane anyway...
I not against more safety and it is probably not a big deal to at least center / throttle down your RC once before flight.
Maybe it is a problem on some PWM ESC that needs to go full throttle and then down for arming ? Not sure if this problem is relevant here.
Why the delay before re-arming is actually needed ?

**Comment by kevindehecker** [comment]
Re-arming delay: #1010 -> after a ardrone2 crash, drone is automatically killed, but at the same time user is trying to kill it ending up in a re-arming.

**Comment by flixr** [comment]
It is fine to merge for me....

**Comment by gautierhattenberger** [comment]
same here

**Comment by flixr** [comment]
merged with 6aa736409887bd4e8be3a6ba1865542a3403aa3d

**Pull request closed without merging**